### PR TITLE
feat: resurrect the nil backend

### PR DESCRIPTION
**Pull request by npmccallum**
This should aid the testing of documentation on systems with no kvm
support (such as GitHub runners).

Signed-off-by: Nathaniel McCallum <nathaniel@profian.com>

### CODE REVIEW

**Comment by npmccallum** [comment]
@haraldh I didn't get this working yet. But if you see something obvious, I'm all ears.

**Comment by haraldh** [comment]
I'll look into this tomorrow.

File: src/backend/nil.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                .at(base + wide.start)
+                .anonymously()
+                .known::<perms::ReadWrite>(Kind::Private)?;
+            eprintln!("map: {:016x}..{:016x}", map.addr(), map.addr() + map.size());
+            map[slim.start - wide.start..][..bytes.len()].copy_from_slice(bytes);
+            forget(
+                map.remap()
+                    .anonymously()
+                    .unknown(Kind::Private, phdr.p_flags as i32)?,
+            );
```
Comment (haraldh) [inline src/backend/nil.rs:92]:
```suggestion
            let mut rwx = 0;

            if phdr.p_flags & PF_R != 0 {
                rwx |= libc::PROT_READ;
            }
            if phdr.p_flags & PF_W != 0 {
                rwx |= libc::PROT_WRITE;
            }
            if phdr.p_flags & PF_X != 0 {
                rwx |= libc::PROT_EXEC;
            }

            if unsafe { libc::mprotect(map.addr() as _, map.size(), rwx) } != 0 {
                return Err(std::io::Error::last_os_error().into());
            }
```

Comment (haraldh) [reply src/backend/nil.rs:92]: This fixed it!

**Review by haraldh — CHANGES_REQUESTED** [review]

**Comment by npmccallum** [comment]
Closed in favor of #1794.

**Pull request closed without merging**